packages/ws/src/Decorators/Broadcast.ts

```diff
@@ -0,0 +1,22 @@
+// broadcast (emit) message to all users on the same namespace (including the peer)
+
+import { initializeMetadata, initializeMetadataMethod } from '@vercube/core';
+import { BaseDecorator, createDecorator } from '@vercube/di';
+
+class BroadcastDecorator extends BaseDecorator {
+
+  public override created(): void {
+    initializeMetadata(this.prototype);
+    const method = initializeMetadataMethod(this.prototype, this.propertyName);
+
+    method.args.push({
```

Comment:
I don’t think this is necessary - we can just override the original method and use the WSS service to handle broadcasting (same for `emit`, `broadcastOthers`):

```ts
public override created(): void {
  const originalMethod = this.instance[this.propertyName];

  this.instance[this.propertyName] = async (...args: any[]) => {
    // execute original method
    const result = await originalMethod.apply(this.instance, args);

    // process result here using WSS service
    this.gWebSocketService.broadcast(result);

    return result;
  }
}
```

In this example, you override the original function and use the result with the WSS service - no need to add extra logic elsewhere.